tests/tests/tier0/monitor-open-close/python/monitor.py

```diff
@@ -0,0 +1,28 @@
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+import sys
+
+from dasbus.error import DBusError
+
+from bluechi.api import Manager, Monitor
+
+# Subscribing to a closed monitor should result in a DBusError since the monitor has removed.
+# In this test we first subscribe to an open monitor to verify everything works, then close
+# the monitor and expect an DBusError when subscribing on a closed monitor.
+
+
+def main():
+    mgr = Manager()
+    monitor_path = mgr.create_monitor()
+    monitor = Monitor(monitor_path=monitor_path)
+    monitor.subscribe('not-important-node', 'not-important-unit')
+    monitor.close()
+    try:
+        monitor.subscribe('not-important-node-2', 'not-important-unit-2')
+    except Exception:
```

Comment:
I know it means bringing in the `unittest` package. But, have you considered using https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaises?

Reply:
it's also supported in pytest (which iirc we already use): https://docs.pytest.org/en/7.1.x/how-to/assert.html#assertions-about-expected-exceptions

Reply:
Refactored it using the `unittest` package. The injected script now doesn't print, but asserts itself and on the calling side we just "forward" the error.